java-client/src/main/java/co/elastic/clients/json/JsonpDeserializerBase.java

```diff
@@ -356,6 +356,43 @@ public Map<String, T> deserialize(JsonParser parser, JsonpMapper mapper, Event e
         }
     }
 
+    // Used for fields such as matched_queries, which can be either an array or a dictionary, where the array
+    // value type is the same as the dictionary key type
+    static class StringArrayMapUnionDeserializer<T> extends JsonpDeserializerBase<Map<String, T>> {
+        private final JsonpDeserializer<T> itemDeserializer;
+
+        protected StringArrayMapUnionDeserializer(JsonpDeserializer<T> itemDeserializer) {
+            super(EnumSet.of(Event.START_OBJECT,Event.START_ARRAY));
+            this.itemDeserializer = itemDeserializer;
+        }
+
+        @Override
+        public Map<String, T> deserialize(JsonParser parser, JsonpMapper mapper, Event event) {
+            Map<String, T> result = new HashMap<>();
+            String key = null;
+            try {
+                // Array case, deserializing it into a map with null values
+                if (event == Event.START_ARRAY) {
+                    while ((event = parser.next()) != Event.END_ARRAY) {
+                        JsonpUtils.ensureAccepts(itemDeserializer, parser, event);
```

Comment:
This should be `JsonpUtils.expectEvent(Event.VALUE_STRING, event)`: `itemDeserializer` is for map values, and in the case of an array we only have string values that would be keys in the map.